santosfamilyfoundation/RuinaMain
# [RUIN-327] Scan license plate

# Description
Current users have to manually enter the car's license plate. To make it easier for the user, and reduce the chance of mistyping the plate number, the app will scan the plate and auto-populate the number in the form.

### Note: Make sure you have the latest version of Node.js to install the license plate scanning library

Follow the instructions on the [Ubuntu formum](https://askubuntu.com/questions/426750/how-can-i-update-my-nodejs-to-the-latest-version) from the third answer (68 votes). If it installs version 17, you will need to change the version to `16.14.2` with the following commands:

`nvm install 16.14.2`
`nvm use 16.14.2`

# Testing

1. Open a report
2. Open the Vehicle form and navigate to "Motor Vehicle License Plate Number"
3. Click on the camera
4. Scan the license plate
5. Check that the populated plate number matches the car
6. Exit the form
7. Re-open the Vehicle Form
8. Check that the data for "Motor Vehicle License Plate Number" was saved
